### PR TITLE
AEC Topologies + Memory Adjustments + Config adjustments

### PR DESCRIPTION
**Pull request by cujomalainey**

### CODE REVIEW

**Comment by cujomalainey** [comment]
There are outstanding bugs in RTNR that this PR breaks regarding period size, so do not merge till we have a fix, I have a local hack I have been using to work around this that I do not plan on upstreaming

File: tools/topology/topology1/sof/pipe-rtnr-google-rtc-audio-processing-capture.m4
Comment (lkoenig) [inline tools/topology/topology1/sof/pipe-rtnr-google-rtc-audio-processing-capture.m4:1]: Could you rename the file 
`pipe-google-rtc-audio-processing-rtnr-capture.m4` to reflect the processing pipeline ?
AEC->RTNR->Capture

File: tools/topology/topology1/CMakeLists.txt
Comment (lkoenig) [inline tools/topology/topology1/CMakeLists.txt:149]: Note: Pipeline name change in the parent commit require a change on that line too.

File: tools/topology/topology1/platform/intel/intel-generic-dmic.m4
Comment (lkoenig) [inline tools/topology/topology1/platform/intel/intel-generic-dmic.m4:49]: This affects only the 16k pipeline. Just want to make sure you are aware of it.
The 48k one uses `DMIC_48k_PERIOD_US` which could be defined in the top topology.
I am wondering if we really need those `INTEL_GENERIC_DMIC_PERIOD` lines depending on `RTNR` being defined.

Comment (cujomalainey) [reply tools/topology/topology1/platform/intel/intel-generic-dmic.m4:49]: Gah thanks for catching this, I must have made the mistake when doing my WIP patch cleanup

File: tools/topology/topology1/sof/pipe-rtnr-google-rtc-audio-processing-capture.m4
Comment (cujomalainey) [reply tools/topology/topology1/sof/pipe-rtnr-google-rtc-audio-processing-capture.m4:1]: ack

**Review by lgirdwood — COMMENTED** [review]
This looks fine to me, but I'm not a Google AEC expert.

**Comment by cujomalainey** [comment]
Please don't merge, just pushed for reviews, only built test so far, ill merge myself if reviews are good

**Comment by lgirdwood** [comment]
> Please don't merge, just pushed for reviews, only built test so far, ill merge myself if reviews are good

Ack - probably bet for @lkoenig or @johnylin76 to approve since they will know the Google code the best.

**Comment by lgirdwood** [comment]
@greg-intel @keqiaozhang @XiaoyunWu6666 fwiw looks like the CI tried to test the USB sound device for this PR ?
https://sof-ci.01.org/sofpr/PR5506/build12490/devicetest/?model=CML_HEL_RT5682&testcase=check-capture-10sec
@cujomalainey I don think this failure is relate to this PR unless something within the CML topology has changed.

**Comment by cujomalainey** [comment]
Only ADL should be affected by this change

File: tools/topology/topology1/platform/intel/intel-generic-dmic.m4
Comment (lkoenig) [inline tools/topology/topology1/platform/intel/intel-generic-dmic.m4:50]: Can we keep the RTNR and DMIC period in the top level topology as it is easier to see if there are inconsistencies at those ones.

**Review by lkoenig — COMMENTED** [review]
The PR looks good. Only the small comment on where should the period for DMIC be defines. I would prefer it to be at top level topology to surface inconsistencies and not override values. That could be the case if one component wants 10ms and the other wants 6ms.

File: tools/topology/topology1/platform/intel/intel-generic-dmic.m4
Comment (cujomalainey) [reply tools/topology/topology1/platform/intel/intel-generic-dmic.m4:50]: @lgirdwood thoughts on this?

Comment (lgirdwood) [reply tools/topology/topology1/platform/intel/intel-generic-dmic.m4]: ack - simplification is good.

File: tools/topology/topology1/CMakeLists.txt
Comment (lgirdwood) [inline tools/topology/topology1/CMakeLists.txt:156]: patch subject style ?

**Review by lgirdwood — COMMENTED** [review]
LGTM

File: tools/topology/topology1/CMakeLists.txt
Comment (cujomalainey) [reply tools/topology/topology1/CMakeLists.txt:156]: Wasn't sure what to put, i am can't reveal the target, I just need this additional config

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt CI stuck here too.

**Comment by wszypelt** [comment]
@lgirdwood Tests run again

**Comment by lgirdwood** [comment]
@cujomalainey fyi - looks like we have some CI issues.

**Pull request closed without merging**
